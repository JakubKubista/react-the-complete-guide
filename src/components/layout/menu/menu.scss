.Menu {
  margin: 0;
  padding: 0;
  list-style: none;
  display: flex;
  align-items: center;
  height: 100%;
}

.Menu li a {
  color: #FFF;
  text-decoration: none;
  height: 100%;
  padding: 16px 10px;
  border-bottom: 4px solid transparent;
  box-sizing: border-box;
  display: block;
}

.Menu li a:hover,
.Menu li a:active,
.Menu li a.active,
.Menu li .Active {
  background-color: #8F5C2C;
  border-bottom: 4px solid #40A4C8;
  color: #FFF;
}