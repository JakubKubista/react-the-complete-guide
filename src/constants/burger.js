export const INGREDIENT_NAMES = {
    breadTop: 'bread-top',
    salad: 'salad',
    cheese: 'cheese',
    meat: 'meat',
    bacon: 'bacon',
    breadBottom: 'bread-bottom'
};

export const INGREDIENT_PRICES = {
    salad: 0.5,
    cheese: 0.4,
    meat: 1.3,
    bacon: 0.7
};

export const CONTROLS = [
    { label: 'Salad', type: INGREDIENT_NAMES.salad },
    { label: 'Cheese', type: INGREDIENT_NAMES.cheese },
    { label: 'Meat', type: INGREDIENT_NAMES.meat },
    { label: 'Bacon', type: INGREDIENT_NAMES.bacon },
];