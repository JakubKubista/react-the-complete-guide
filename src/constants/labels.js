export const MESSAGES = {
  burgerBuilder: 'Burger Builder',
  noIngredients: 'Please start adding ingredients.',
  ingredientsNotLoaded: 'Ingredients cannot be loaded!',
  enterContactData: 'Enter your Contact Data',
  enterValidValue: 'Please enter a valid value',
  enjoyYourMeal: 'We hope it tastes well!',
  yourOrder: 'Your Order',
  burgerWithIngredients: 'A delicious burger with the following ingredients:',
  totalPrice: 'Total Price',
  continueToCheckout: 'Continue to Checkout?',
  ingredients: 'Ingredients',
  price: 'Price',
  currencyUsd: 'USD',
  loading: 'Loading...'
};

export const BUTTONS = {
  order: 'ORDER',
  orderNow: 'ORDER NOW',
  submit: 'SUBMIT',
  cancel: 'CANCEL',
  continue: 'CONTINUE',
  less: 'Less',
  more: 'More'
};
