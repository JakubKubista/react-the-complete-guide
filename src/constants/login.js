export const LOGIN_FORM = {
  email: {
    elementType: 'input',
    elementConfig: {
      type: 'email',
      placeholder: 'E-mail'
    },
    value: '',
    validation: {
      required: true
    },
    valid: false,
    touched: false
  },
  password: {
    elementType: 'input',
    elementConfig: {
      type: 'password',
      placeholder: 'Password'
    },
    value: '',
    validation: {
      required: true,
      minLength: 6
    },
    valid: false,
    touched: false
  },
}