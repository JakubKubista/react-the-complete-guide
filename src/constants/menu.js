export const MENU_ICON = {
  src: 'https://icon-library.net/images/white-hamburger-menu-icon/white-hamburger-menu-icon-24.jpg',
  style: {
    height: '64px',
    marginTop: '6px'
  },
  alt: 'icon-menu'
};

export const MENU_ELEMENTS = {
  labels: {
    home: 'Burger Builder',
    orders: 'Orders',
    auth: 'Login'
  },
  routes: {
    home: '/',
    orders: '/orders',
    auth: '/auth'
  }
};
