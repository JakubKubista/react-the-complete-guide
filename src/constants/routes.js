export const ROUTES = {
  home: '/',
  orders: '/orders',
  signIn: '/signin',
  signOut: '/signout',
  checkout: '/checkout'
};