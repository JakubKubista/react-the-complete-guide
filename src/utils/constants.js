export const API_URL = 'https://react-hooks-update-2cb33.firebaseio.com/ingredients.json';