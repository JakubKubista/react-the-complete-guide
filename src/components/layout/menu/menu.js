import React from 'react';
import classes from './menu.scss';
import { MENU_ITEMS } from '../../../constants/menu';
import MenuItem from './menu-item/menu-item';

const Menu = (props) => {
  const menuItems = [];
  for (var name in MENU_ITEMS) {
    let element = MENU_ITEMS[name];

    if (name === 'auth') {
      element = props.isSignedIn ? MENU_ITEMS[name].signOut : MENU_ITEMS[name].signIn;
    }

    menuItems.push(
      <MenuItem
        key={element.route}
        link={element.route}
        exact>
          {element.label}
      </MenuItem>
    );
  }

  return (
    <ul className={classes.Menu}>
      {menuItems}
   </ul>
  )
};

export default Menu;