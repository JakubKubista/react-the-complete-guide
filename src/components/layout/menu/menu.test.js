import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { MENU_ITEMS } from '../../../constants/menu';
import Menu from './menu';
import MenuItem from './menu-item/menu-item';

configure({adapter: new Adapter()});

describe('<Menu/>', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<Menu />);
  })

  it('should render 2x <MenuItem /> if isSignedIn is not true', () => {
    expect(wrapper.find(MenuItem)).toHaveLength(2);
  });

  it('should render 3x <MenuItem /> if isSignedIn is true', () => {
    wrapper.setProps({isSignedIn: true});

    expect(wrapper.find(MenuItem)).toHaveLength(3);
  });

  it('should have <MenuItem /> with Sign In', () => {
    expect(wrapper.contains(
      <MenuItem link={MENU_ITEMS.auth.signIn.route} exact>
        {MENU_ITEMS.auth.signIn.label}
      </MenuItem>
    )).toEqual(true);
  });
  it('should have <MenuItem /> with Sign Out', () => {
    wrapper.setProps({isSignedIn: true});

    expect(wrapper.contains(
      <MenuItem link={MENU_ITEMS.auth.signOut.route} exact>
        {MENU_ITEMS.auth.signOut.label}
      </MenuItem>
    )).toEqual(true);
  });
});