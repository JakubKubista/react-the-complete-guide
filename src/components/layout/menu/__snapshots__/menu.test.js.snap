// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> should match to snapshot 1`] = `
<ul>
  <li>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Burger Builder
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      href="/signin"
      onClick={[Function]}
    >
      Sign In
    </a>
  </li>
</ul>
`;
